conans/test/utils/mocks.py

```diff
@@ -62,6 +62,9 @@ def __getattr__(self, name):
         except KeyError:
             raise ConanException("'%s' value not defined" % name)
 
+    def rm_safe(self, name):
+        self.values.pop(name)
```

Comment:
Maybe 
```suggestion
        self.values.pop(name, None)
```
so it matches the new behaviour and does not raise when the code wouldnt either